playbook_ui/lib/src/playbook_gallery.dart

```diff
@@ -23,22 +25,24 @@ class PlaybookGallery extends StatefulWidget {
 }
 
 class _PlaybookGalleryState extends State<PlaybookGallery> {
-  final _textEditingController = TextEditingController();
+  late final TextEditingController _defaultSearchTextController = TextEditingController()
+    ..addListener(_searchTextListener);
+  TextEditingController get _effectiveSearchTextController =>
+      widget.searchTextController ?? _defaultSearchTextController;
+
   final _scrollController = ScrollController();
   List<Story> _stories = [];
 
   @override
   void initState() {
     super.initState();
     _updateStoriesFromSearch();
-    _textEditingController.addListener(() {
-      setState(_updateStoriesFromSearch);
-    });
   }
 
   @override
   void dispose() {
-    _textEditingController.dispose();
+    widget.searchTextController?.removeListener(_searchTextListener);
+    _defaultSearchTextController.dispose();
```

Comment:
Shouldn't it be checking for null here?
```suggestion
    if (widget.searchTextController == null) {
      _defaultSearchTextController.dispose();
    }
```

Reply:
`_defaultSearchTextController` is hold by our own state,
so any way we should dispose it when the disposing state.

Reply:
Does exist instance of `_defaultSearchTextController` whether `widget.searchText Controller` is `null` or not? 
If so, no problem.

Reply:
_defaultSearchTextController is late final, so anyway disposed it should be safer.